src/sssom/cli.py

```diff
@@ -660,14 +661,20 @@ def sort(input: str, output: TextIO, by_columns: bool, by_rows: bool):
 #     write_table(msdf=filtered_msdf, file=output)
 
 
-def dynamically_generate_sssom_options(options) -> Callable[[Any], Any]:
+P = ParamSpec("P")
+T = TypeVar("T")
+
+
+def dynamically_generate_sssom_options(
+    options: List[str],
```

Comment:
Minor nitpick: Some type hints use `List[...]` (as on this line), whereas some other type hints use `list[...]`.

My understanding is that the capitalized version is only required for Python 3.8 and earlier, while Python 3.9+ can accommodate both forms (capitalized or not).

Since we are already no longer supporting Python 3.8, maybe we could uniformly use the lowercase forms, instead of a mix of both forms?